many: do a mechanical origin->developer rename

mvo5 opened the pull request:
We do not longer want to use "origin" and use "developer" instead.

mvo5 commented:
retest this please

chipaca commented:
:+1:

zyga commented:
+1, though let's wait for the integration tests to go green.

niemeyer commented on docs/security.md:
```diff
@@ -37,8 +37,8 @@ service/binary name and package version. The `APP_ID` takes the form of
-and the app was uploaded to the `myorigin` origin in the store, then the
-`APP_ID` for the `bar` service is `foo.myorigin_bar_0.1`. The `APP_ID` is used
+and the app was uploaded to the `mydeveloper` developer in the store, then the
+`APP_ID` for the `bar` service is `foo.mydeveloper_bar_0.1`. The `APP_ID` is used
```
s/mydeveloper/snapdev/, probably worth as a mechanical change to catch all occurrences.

niemeyer commented on snap/remote/remote.go:
```diff
@@ -47,4 +46,8 @@ type Snap struct {
+
+	// FIXME: the store should return "developer" to us instead of
+	//        origin
+	Developer string `json:"origin" yaml:"origin"`
```
Is the "yaml" here necessary?  Do we use this type to sync from disk or anywhere else besides the store?

One alternative would be to keep the Origin field unchanged, and introduce the Developer field that is manually populated on queries from the store. Best approach depends on how we use this.

mvo5 replied:
We currently use this "developer" yaml in the snap/removed/removed.go code. As soon as this code dies we don't need the yaml anymore. As soon as the store provides as with "developer" instead of "origin" it will also work naturally (except for upgrade concerns that we might ignore because we are still on rolling).

niemeyer commented on snappy/overlord.go:
```diff
@@ -381,8 +381,8 @@ func (o *Overlord) partsForGlobExpr(globExpr string) (parts []*Snap, err error)
 
-		origin, _ := originFromYamlPath(realpath)
-		snap, err := NewInstalledSnap(realpath, origin)
+		developer, _ := developerFromYamlPath(realpath)
```
These "from path" methods will soon need to go away.

mvo5 replied:
Indeed, they will go away naturally because the "developer" will no longer be used in the path.

niemeyer commented:
The mechanical change went really well. Have a couple of minor comments.

LGTM

mvo5 commented:
@niemeyer Thanks a bunch and good catch(es) on the open issues. I addressed the concerns and answered the question. Please let me know if there is more that should be done here.
